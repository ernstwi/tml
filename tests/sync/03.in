init foo
    invar { x < 5 }

foo -> bar
    guard { x == 10 }

bar -> baz
    guard { x < 3 & y > 5 }
