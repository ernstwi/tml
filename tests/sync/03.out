{
    "channels": [],
    "clocks": [
        "x",
        "y"
    ],
    "edges": [
        {
            "from": "bar",
            "guard": [
                {
                    "clock": "x",
                    "clocks_n": 1,
                    "cmp": "lt",
                    "n": 3
                },
                {
                    "clock": "y",
                    "clocks_n": 1,
                    "cmp": "gt",
                    "n": 5
                }
            ],
            "reset": null,
            "sync": null,
            "to": "baz"
        },
        {
            "from": "foo",
            "guard": [
                {
                    "clock": "x",
                    "clocks_n": 1,
                    "cmp": "eq",
                    "n": 10
                }
            ],
            "reset": null,
            "sync": null,
            "to": "bar"
        }
    ],
    "locations": [
        {
            "id": "bar",
            "initial": false,
            "invariant": null
        },
        {
            "id": "baz",
            "initial": false,
            "invariant": null
        },
        {
            "id": "foo",
            "initial": true,
            "invariant": [
                {
                    "clock": "x",
                    "cmp": "lt",
                    "n": 5
                }
            ]
        }
    ]
}
